src/runtime/storage.ts

```diff
@@ -101,12 +103,39 @@ export class Storage {
 
   /**
    * Retrieves timestamps from local storage, under the storageKey provided.
-   * Filters out timestamps older than a week.
+   * Filters out timestamps older than a week. To be deprecated in favor of new
+   * getFrequencyCappingEvent with new frequency capping flow.
    */
   async getEvent(storageKey: string): Promise<number[]> {
     const value = await this.get(storageKey, /* useLocalStorage */ true);
     return this.pruneTimestamps_(this.deserializeTimestamps_(value));
   }
+  /**
+   * Stores the current timestamp to local storage, under the storageKey
+   * provided for the frequency capping flow. To replace the legacy storeEvent
+   * fn when the legacy triggering flow is deprecated. Prunes timestamps
+   * older than two weekw in the process.
+   */
+  async storeFrequencyCappingEvent(storageKey: string): Promise<void> {
+    const timestamps = await this.getFrequencyCappingEvent(storageKey);
+    timestamps.push(Date.now());
+    const valueToStore = this.serializeTimestamps_(timestamps);
+    this.set(storageKey, valueToStore, /* useLocalStorage */ true);
+  }
+
+  /**
+   * Retrieves timestamps from local storage, under the storageKey provided
+   * for the frequency capping flow. To replace the legacy getEvent fn when the
+   * legacy triggering flow is deprecated. Prunes timestamps older than two
+   * weeks.
+   */
+  async getFrequencyCappingEvent(storageKey: string): Promise<number[]> {
+    const value = await this.get(storageKey, /* useLocalStorage */ true);
+    return this.pruneTimestamps_(
+      this.deserializeTimestamps_(value),
+      TWO_WEEKS_IN_MILLIS
```

Comment:
Can you add a comment that 2 weeks is not a technical limitation, but based on the product requirement?